python/ray/tune/suggest/ax.py

```diff
@@ -190,20 +185,26 @@ def _setup_experiment(self):
                 objective_name=self._metric,
                 parameter_constraints=self._parameter_constraints,
                 outcome_constraints=self._outcome_constraints,
-                minimize=self._mode != "max")
+                minimize=self._mode != "max"
+            )
         else:
             if any([
                     self._space, self._parameter_constraints,
-                    self._outcome_constraints
+                    self._outcome_constraints, self._mode, self._metric
             ]):
                 raise ValueError(
                     "If you create the Ax experiment yourself, do not pass "
                     "values for these parameters to `AxSearch`: {}.".format([
-                        "space", "parameter_constraints", "outcome_constraints"
+                        "space", 
+                        "parameter_constraints", 
+                        "outcome_constraints",
+                        "mode",
+                        "metric",
                     ]))
 
         exp = self._ax.experiment
-        self._objective_name = exp.optimization_config.objective.metric.name
```

Comment:
I changed this from setting `self._objective_name` to setting `self._metric`, because it felt like that was the intention here –– please correct if I was wrong : )

Reply:
Ah yes, I think this might be correct, except if Ax internally uses a different representation (e.g. with a prefix or so) - I'd have to look it up. If tests pass like this, it's probably ok